css-naming-convention.md

```diff
@@ -0,0 +1,195 @@
+
+And now let's break it down into all the different parts.
+
+```html
+<button class='
+  namespace-ComponentName-childElement 
+  namespace-ComponentName-childElement--modifier
+  is-state
+'>
+```
+
+
+### Namespace
+
+`github`-CommitBox
+
+Every class starts with a namespace, in Atom's case it's the __package name__. Since packages are unique (well, at least for all the packages published on atom.io), it avoids style conflicts. Using a namespace like this makes sure that no styles __leak out__. And even more importantly that no styles __leak in__.
+
+The namespace for Atom core ([atom/atom](https://github.com/atom/atom-ui)) is `core`-ComponentName.
+
+Atom's [UI library](https://github.com/atom/atom-ui) is the only exception that doesn't use a namespace. All components start with the `ComponentName`. For example `Button`, `Checkbox`.
+
+> Note: If multiple words are needed, camelCase is used: `myPackage`-Component.
+
+
+### Component
+
+github-`CommitBox`
+
+Components are building blocks of a package. They can be small or large. Components can also contain other components. It's more about seeing what belongs together.
+
+Components can also share the same elements. A pattern often found is that a new Component starts where a childElement ends. 
+
+```html
+<ul class="github-List">
+  <li class="github-List-item github-Commit">
+    <label class="github-Commit-message"></label>
+    <span class="github-Commit-time"></span>
+  </li>
+</ul>
+```
+
+In this example, `github-List-item` is responsible for the "container" and layout styles. `github-Commit` is responsible for the "content" inside.
+
+> Note: Components use PascalCase. This makes it easy to spot them in the markup. For example in `settings-List-item`, it's easy to see that the component is `List` and not `settings` or `item`.
+
+
+### Child element
+
+github-CommitBox-`commitButton`
+
+Elements that are part of a component are appended to the component's class name with a single dash (`-`).
+
+> Note: If multiple words are needed, camelCase is used: github-CommitBox`-commitButton`.
+
+
+### Modifier
+
+github-CommitBox-commitButton`--primary`
+
+Modifiers are used if a component is very similar to the default component but varies slightly. Like has a different color. Modifiers use a double dash `--` to distinguish them from components and child elements.
+
+> Note: If multiple words are needed, camelCase is used: github-CommitBox-commitButton`--primaryColor`.
+
+
+### States
+
+github-CommitBox-commitButton `is-disabled`
+
+States are prefixed with a short verb, like `is-` or `has-`. Since these class names probably get used in many different places, it should never be styled stand-alone and always be a chained selector.
+
+```less
+.is-disabled {
+  // nope
+}
+.github-CommitBox-commitButton.is-disabled {
+  // yep
+}
+```
+
+> Note: If multiple words are needed, camelCase is used: `has-collapsedItems`.
+
+
+
+## More guidelines, not crucial, but still good to know and follow. ;-)
```

Comment:
I'd probably shorten this to just "More guidelines" if this is going to be a formal document 😉.